core/app/models/spree/carton.rb

```diff
@@ -3,7 +3,7 @@
 class Spree::Carton < Spree::Base
   belongs_to :address, class_name: 'Spree::Address', optional: true
   belongs_to :stock_location, class_name: 'Spree::StockLocation', inverse_of: :cartons, optional: true
-  belongs_to :shipping_method, class_name: 'Spree::ShippingMethod', inverse_of: :cartons, optional: true
+  belongs_to :shipping_method, -> { with_deleted }, class_name: 'Spree::ShippingMethod', inverse_of: :cartons, optional: true
```

Comment:
I think we should use `with_discarded` instead

Reply:
Done. I force-pushed this change.